src/services/callHierarchy.ts

```diff
@@ -129,6 +129,31 @@ namespace ts.CallHierarchy {
         return { text, pos: declName.getStart(), end: declName.getEnd() };
     }
 
+    function getCallHierarchItemContainerName(node: CallHierarchyDeclaration): string | undefined {
+        if (isConstNamedExpression(node)) {
+            if (isModuleBlock(node.parent.parent.parent.parent) && isIdentifier(node.parent.parent.parent.parent.parent.name)) {

```

Comment:
Is it safe to do this many unguarded parent accesses?

Reply:
Yes—`isConstNamedExpression` guarantees that the third-to-last `.parent` is a `VariableDeclarationList` or `CatchClause`, so the second-to-last is `VariableStatement | ForStatement | ForOfStatement | ForInStatement | TryStatement`, so one more `.parent` is safe from there.